src/extension.rs

```diff
@@ -7,6 +7,29 @@
 use crate::error::{NiftiError, Result};
 use byteordered::{ByteOrdered, Endian};
 use std::io::{ErrorKind as IoErrorKind, Read};
+use num_derive::FromPrimitive;
+
+/// Data type for representing a NIFTI ecodes
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, FromPrimitive)]
+#[repr(u32)]
+pub enum NiftiEcode {
+    /// Ignore the extension
+    NiftEcodeIgnore = 0,
+    /// DICOM 
+    NiftiEcodeDicom = 2,
+    /// AFNI extension in XML format
+    NiftiEcodeAFNI = 4,
+    /// String Comment
+    NiftiEcodeComment = 6,
+    /// XCEDE extension in XML format
+    NiftiEcodeXCEDE = 8,
+    /// JimDimInfo
+    NiftiEcodeJimDimInfo = 10,
+    /// WorkflowFWDS
+    NiftiEcodeWorkflowFWDS = 12,
+    /// Freesurfer
+    NiftiEcodeFreesurfer = 14,
```

Comment:
What is your source of information for those constants? I usually use [this page](https://brainder.org/2012/09/23/the-nifti-file-format/) for everything related to nifti and I can only see 0, 2 and 4.

Reply:
https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1_io.h
actually goes to 44 now.